File: oop/lab3/cpp/src/hexagon.cpp
#include <stdexcept>

#include "hexagon.hpp"

Hexagon::Hexagon(double s) noexcept(false)
{
    if (s <= 0)
    {
        throw std::invalid_argument("Side length must be greater than 0.");
    }
    this->side = s;
}

double Hexagon::area() const noexcept(true)
{
    return 3 * sqrt(3) * this->side * this->side / 2;
}

double Hexagon::perimeter() const noexcept(true)
{
    return 6 * this->side;
}

const char *Hexagon::name() const noexcept(true)
{
    return "Hexagon";
}

File: oop/lab3/java/Hexagon.java
public class Hexagon implements Figure {
    private double side;

    public Hexagon(double s) throws IllegalArgumentException {
        if (s <= 0) {
            throw new IllegalArgumentException("Side length must be positive.");
        }
        this.side = s;
    }

    @Override
    public double area() {
        return (3 * Math.sqrt(3) * side * side) / 2;
    }

    @Override
    public double perimeter() {
        return 6 * side;
    }

    @Override
    public String name() {
        return "Hexagon";
    }
}
